### ISSUE TITLE
Pyright ignores configuration from custom toml files

### ISSUE THREAD

**Issue by vprud** [microsoft/pyright#7972]
**Describe the bug**

Option `--project` of command-line version of pyright does not support configuration in toml file. The configuration in toml files other than 'pyproject.toml` is ignored. If emit verbose diagnostics of pyright, it will show that it does not find any configuration file. 

**Code or Screenshots**

For example, I will use `configs` folder, in which configuration files will be saved. Create configuration in `configs/pyright.toml`:
```toml
[tool.pyright]
typeCheckingMode = "strict"
```
and similarly in `configs/pyright.json`:
```json
{
    "typeCheckingMode": "strict"
}
```

When running with toml configuration file:
```shell
pyright --project configs/pyright.toml --verbose
```
pyright writes that it cannot find configuration file:
```shell
Configuration file not found at <path to root>/configs/pyright.toml.
```

When running with json configuration file:
```shell
pyright --project configs/pyright.json --verbose
```
pyright writes that it found configuration file in `./configs/pyright.json` and uploaded it:
```shell
Loading configuration file at <path to root>/configs/pyright.json
```

This behavior does not seem consistent for different formats of configuration file. It is expected that pyright will read toml file as well as  json configuration  file.

**VS Code extension or command-line**
```
$ pyright --version
pyright 1.1.360
```

**Comment by erictraut** [comment]
This is by design. If you want to use a custom configuration, you should use the ".json" format. Support for ".toml" is provided only for integration with the standard "pyproject.toml" file that is commonly used for Python projects.

**Comment by erictraut** [comment]
Reopening this as a feature request.

**Comment by erictraut** [comment]
I decided to implement this feature as part of #4366. It will be included in the next release.

**Comment by vprud** [comment]
Great, I believe it will be very useful.

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.
